core/data_category.js

```diff
@@ -439,6 +465,18 @@ Blockly.DataCategory.addBlock = function(xmlList, variable, blockType,
   }
 };
 
+/**
+ * Construct a flyout label with the given text tag. Add the label to the given
+ *     xmlList.
+ * @param {!Array.<!Element>} xmlList Array of XML block elements.
+ * @param {string} text Text content of the label.
+ */
+Blockly.DataCategory.addLabel = function(xmlList, text) {
+  var labelText = '<xml><label text="' + text + '"></label></xml>';
```

Comment:
This feels vulnerable to nonsense similar to XSS, e.g. if a translator put `"></label>` in the translation string. Is `setAttribute` better here, i.e. `labelText = '<xml></label></label></xml>'` and `label.setAttribute('text', text)`?